nowcasting_dataset/data_sources/gsp/gsp_data_source.py

```diff
@@ -399,14 +416,54 @@ def drop_gsp_by_threshold(gsp_power: pd.DataFrame, meta_data: pd.DataFrame, thre
 
     keep_index = maximum_gsp > threshold_mw
 
-    logger.debug(f"Dropping {sum(~keep_index)} GSPs as maximum is not greater {threshold_mw} MW")
+    dropped_gsp_names = meta_data.loc[keep_index[~keep_index].index].gsp_name
+
+    logger.debug(
+        f"Dropping {sum(~keep_index)} GSPs as maximum is not greater {threshold_mw} MW."
+        f" Dropped GSP IDs and GSP names:\n{dropped_gsp_names}"
+    )
     logger.debug(f"Keeping {sum(keep_index)} GSPs as maximum is greater {threshold_mw} MW")
 
-    gsp_power = gsp_power[keep_index.index]
-    gsp_ids = gsp_power.columns
-    meta_data = meta_data[meta_data["gsp_id"].isin(gsp_ids)]
+    filtered_gsp_power = gsp_power.loc[:, keep_index]
+    filtered_gsp_ids = filtered_gsp_power.columns
+    filtered_meta_data = meta_data[meta_data["gsp_id"].isin(filtered_gsp_ids)]
+
+    assert set(filtered_gsp_power.columns) == set(filtered_meta_data.gsp_id)
+    return filtered_gsp_power, filtered_meta_data
+
+
+def drop_gsp_north_of_boundary(
```

Comment:
I know I sound like a bit of a broken record. But could you make an issue for writing a unittest test for this function? 
Or this morning I might push one to this branch if thats ok

Reply:
Good idea!  I've added an issue here: #503